project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -137,6 +145,18 @@ PlaceDetails getPlaceDetails(PlaceDetailsRequest request)
     }
 
     private static String createCuisinesQuery(ImmutableList<String> cuisines) {
-        return String.join("|", cuisines);
+        return cuisines.stream()
+            .map(cuisine -> String.join("|", CUISINE_TO_SEARCH_WORDS_MAP.get(cuisine)))
+            .collect(Collectors.joining("|"));
+    }
+
+    private static Map<String, String[]> getCuisinesMap() {
+        System.out.println(PlacesFetcher.class.getClassLoader());
+        Type mapType = new TypeToken<Map<String, String[]>>() {
+        }.getType();
+        return new Gson().fromJson(new JsonReader(
+            new InputStreamReader(
+                PlacesFetcher.class.getResourceAsStream("cuisinesSearchWords.json"))),
```

Comment:
Keep this in a named static final.

Reply:
The getCuisineMap() is only called once to generate the  private static final CUISINE_TO_SEARCH_WORDS map. So I think it will not spare calls for new InputStreamReader() to keep it as a static field.
Or did you mean to give it a name inside this function for better readability?

Reply:
I meant having the string "cuisinesSearchWords.json" as a static final.

Reply:
Ok, changed it